rust/src/partitions.rs

```diff
@@ -0,0 +1,101 @@
+use std::convert::TryFrom;
+
+use crate::DeltaTableError;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum PartitionValue<T> {
+    Equal(T),
+    NotEqual(T),
+    In(Vec<T>),
+    NotIn(Vec<T>),
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct PartitionFilter<'a, T> {
```

Comment:
typically in query engine implementations, predicates/filters are modeled as structs with 3 fields `key, op, value` or `left, op, right`. we are merging `op` with `value/right` here :)

but i think this is something minor we can refactor later, no need to block the merge.

Reply:
Interesting, let me know if I need to change it!

Reply:
I don't think we need to refactor this in this PR since I think it's something minor and I have already approved it : ) But it is definitely a good change to introduce in future follow ups.